db/db_impl/db_impl_open.cc

```diff
@@ -1545,6 +1545,70 @@ Status DB::Open(const DBOptions& db_options, const std::string& dbname,
                       !kSeqPerBatch, kBatchPerTxn);
 }
 
+// TODO: Implement the trimming in flush code path.
+// TODO: Perform trimming before inserting into memtable during recovery.
+// TODO: Pick files with max_timestamp > trim_ts and compress it.
+Status DB::OpenAndTrimHistory(
+    const DBOptions& db_options, const std::string& dbname,
+    const std::vector<ColumnFamilyDescriptor>& column_families,
+    std::vector<ColumnFamilyHandle*>* handles, DB** dbptr,
+    const std::string trim_ts) {
+  assert(dbptr != nullptr);
+  assert(handles != nullptr);
+  auto validateOptions = [&] {
+    if (db_options.avoid_flush_during_recovery) {
+      return Status::InvalidArgument(
```

Comment:
Can we add a unit test for this, just as sanity check?

Reply:
Add OpenAndTrimHistoryInvalidOptionTest